alacritty/src/event.rs

```diff
@@ -651,13 +647,13 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         let timer_id = TimerId::new(Topic::BlinkCursor, self.display.window.id());
         if self.scheduler.unschedule(timer_id).is_some() {
             self.schedule_blinking();
-            self.display.cursor_hidden = false;
+            if mem::replace(&mut self.display.cursor_hidden, false) {
+                *self.dirty = true;
+            }
```

Comment:
I'd add an empty line above and put a quick comment that we only need to redraw if the cursor is currently hidden.

Also since this is a `bool` you can use `mem::take` instead.